src/codeu/chat/client/commandline/Chat.java

```diff
@@ -457,4 +530,371 @@ public void invoke(List<String> args) {
     // so that it can be used.
     return panel;
   }
+  
+  private Panel createInterestPanel(final UserContext user) {
+
+    final Panel panel = new Panel();
+    
+    InterestSet interests = context.getInterestSet(user.user.id);
+    int updates = 0;
+    
+    ArrayList<Bookmark> display = new ArrayList<Bookmark>();
+    ArrayList<Time> unsorted = new ArrayList<Time>();
+    
+    for (User u : interests.users){
+      HashSet<ConversationContext> newSet = conversationsOfUser(u,user);
+      
+      for(ConversationContext c : newSet){
+        boolean isCopy = false;
+        for(Bookmark b : interests.bookmarks){
+          if(!isCopy){
+            if(b.conversation.title.equals(c.conversation.title)){
+          	  isCopy = true;
+            }
+          }
+        }
+        
+        if(!isCopy){
+          unsorted.add(c.conversation.creation);
+          display.add(new Bookmark(c));
+          interests.bookmarks.add(new Bookmark(c));
+        }
+          
+        }
+       }
+       
+    if (unsorted.size() > 0){
+	  Mergesort merge = new Mergesort();
+      display = merge.sort(unsorted, display);
+    }
+       
+       
+    for(Bookmark b : display){
+    	ConversationContext c = findConversation(b.conversation.title, user);
+        System.out.println("--- new conversation "+c.conversation.title+" from "+findUsername(c.conversation.owner)+" ---");
+          if(c.firstMessage() != null){
+            for (MessageContext message = c.firstMessage();
+                           message != null;
+                           message = message.next()) {
+              System.out.println();
+              System.out.format("USER : %s\n", findUsername(message.message.author));
```

Comment:
This code that displays a list of bookmarks is repeated. Can you refactor it out into a separate method, perhaps called something like `displayBookmarks(List<Bookmark> bookmarks)` so that you only have to write this logic once?